src/assets/TokenRatesController.ts

```diff
@@ -315,27 +345,79 @@ export class TokenRatesController extends BaseController<
 
     const slug = await this.getChainSlug();
-
     const newContractExchangeRates: ContractExchangeRates = {};
     if (!slug) {
       this.tokenList.forEach((token) => {
         const address = toChecksumHexAddress(token.address);
         newContractExchangeRates[address] = undefined;
       });
     } else {
+      // check if native currency is supported as a vs_currency by the API
+      const nativeCurrencySupported = await this.checkIsSupportedVsCurrency(
+        nativeCurrency,
+      );
+      const vsCurrency = nativeCurrencySupported
+        ? nativeCurrency.toLowerCase()
+        : FALL_BACK_VS_CURRENCY.toLowerCase();
       const pairs = this.tokenList.map((token) => token.address).join(',');
-      const query = `contract_addresses=${pairs}&vs_currencies=${nativeCurrency.toLowerCase()}`;
+      const query = `contract_addresses=${pairs}&vs_currencies=${vsCurrency}`;
       const prices = await this.fetchExchangeRate(slug, query);
+      let updatedPrices = prices;
+      if (!nativeCurrencySupported) {
+        updatedPrices = await this._updateConversionRates(
+          prices,
+          nativeCurrency,
+        );
+      }
       this.tokenList.forEach((token) => {
         const address = toChecksumHexAddress(token.address);
-        const price = prices[token.address.toLowerCase()];
+        const price = updatedPrices[token.address.toLowerCase()];
         newContractExchangeRates[address] = price
           ? price[nativeCurrency.toLowerCase()]
           : 0;
       });
     }
     this.update({ contractExchangeRates: newContractExchangeRates });
   }
+
+  /**
+   * Updates the conversion rates from from token/eth to token/nativeCurrency
+   * if 'eth' was used as fallback vscurrency for querying because the nativeCurrency isn't supported
+   *
+   * @param prices - the object with conversion rates returned by coingecko with token/eth rates
+   * @param nativeCurrency - the native currency of the currently active network
+   * @returns - Promise resolving to an object with conversion rates for each token
+   * related to the network's native currency
+   */
+  async _updateConversionRates(
+    prices: CoinGeckoResponse,
+    nativeCurrency: string,
+  ): Promise<CoinGeckoResponse> {
+    let nativeCurrencyConversionRate = 0;
+    try {
+      ({
+        conversionRate: nativeCurrencyConversionRate,
+      } = await fetchNativeExchangeRate(
+        FALL_BACK_VS_CURRENCY,
+        nativeCurrency,
+        false,
+      ));
+    } catch (error) {
+      if (!error.message.includes('market does not exist for this coin pair')) {
+        throw error;
+      }
+    }
+
+    for (const [tokenAddress, conversion] of Object.entries(prices)) {
+      const ethConversionRate = conversion[FALL_BACK_VS_CURRENCY.toLowerCase()];
+      // convert from token/eth to token/nativeCurrency
+      prices[tokenAddress] = {
+        [nativeCurrency.toLowerCase()]:
+          ethConversionRate / nativeCurrencyConversionRate,
```

Comment:
`nativeCurrencyConversionRate` defaults to zero, so we'd be dividing by zero here in the "market does not exist for this coin pair" case :thinking:

Reply:
Good catch. If we get `market does not exist for this coin pair` (or any error from the API that results in no conversion rate) we probably want to eject from the process of updating exchange rates since they won't be valid. I suppose either this:
```
 try {
      ({
        conversionRate: nativeCurrencyConversionRate,
      } = await fetchNativeExchangeRate(
        FALL_BACK_VS_CURRENCY,
        nativeCurrency,
        false,
      ));
    } catch (error) {
      if (error.message.includes('market does not exist for this coin pair')) {
        return {};
      } else {
        throw error;
      }
    }
  ```
  or we don't wrap it in a try/catch at all.
  
  What do you think?

Reply:
Throwing an error is probably best, so the caller can decide how to handle the failure.

What was the reason for suppressing `market does not exist for this coin pair` again? I think I knew this at one point but I must have forgotten.

Reply:
Well it just means that the API isn't aware of one or both of the token's whose prices are being queried about. Really they shouldn't send this back as an error but its what we've got.

Reply:
Oh OK, right. So in that case we want to set the conversion rate to zero I guess? I think that is what we do at the moment. That sounds reasonable.

Reply:
Yes exactly which this 
```
 if (error.message.includes('market does not exist for this coin pair')) {
        return {};
      }
```
results in because when 
```
  updatedPrices = await this._updateConversionRates(
          prices,
          nativeCurrency,
```
returns `{ }`

price will be undefined here 
```
 const price = updatedPrices[token.address.toLowerCase()];
        newContractExchangeRates[address] = price
          ? price[nativeCurrency.toLowerCase()]
          : 0;
```
and so the conversionRate gets set to `0`